### PR TITLE
Change the generator script to kube_codegen.sh

### PR DESCRIPTION
**Pull request by TaiPark**
As https://github.com/volcano-sh/apis/issues/133 suggested, try to fix the current code generator issue by replacing the current codegen script with `kube_codegen.sh`.

It appears that the change has introduced a set of `Apply` methods, please confirm if this is as expected :)

### CODE REVIEW

**Comment by Monokaix** [comment]
All the new generated files should have a header of year 2024, maybe you can modify the template file hack/boilerplate.go.txt to achieve this.

**Comment by TaiPark** [comment]
> All the new generated files should have a header of year 2024, maybe you can modify the template file hack/boilerplate.go.txt to achieve this.

Done.

**Comment by Monokaix** [comment]
I found another problem, if we change the year to 2024, the original files that already have year will also change to 2024, which is unacceptable, and according to the issue https://github.com/kubernetes/gengo/issues/24, we can directly remove the year field in license header template.

**Comment by TaiPark** [comment]
> I found another problem, if we change the year to 2024, the original files that already have year will also change to 2024, which is unacceptable, and according to the issue [kubernetes/gengo#24](https://github.com/kubernetes/gengo/issues/24), we can directly remove the year field in license header template.

Done.

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -20,22 +20,20 @@ set -o pipefail
...
-bash ${SCRIPT_ROOT}/hack/generate-internal-groups.sh "deepcopy,conversion,openapi" \
-  volcano.sh/apis/pkg/apis/ volcano.sh/apis/pkg/apis volcano.sh/apis/pkg/apis \
-  "batch:v1alpha1 scheduling:v1beta1"   \
-  --output-base "$(dirname "${BASH_SOURCE[0]}")/../../.." \
-  --go-header-file ${SCRIPT_ROOT}/hack/boilerplate.go.txt
+kube::codegen::gen_helpers \
+    --boilerplate "${SCRIPT_ROOT}/hack/boilerplate.go.txt" \
+    "${SCRIPT_ROOT}/pkg/apis"
 
+kube::codegen::gen_client \
```
Comment (Monokaix) [inline hack/update-codegen.sh:33]: Can we also generate openapi?

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**